dttools/src/priority_queue.h

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright (C) 2024 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file COPYING for details.
+*/
+
+#ifndef PRIORITY_QUEUE_H
+#define PRIORITY_QUEUE_H
+
+/** 
+@file priority_queue.h
+A general purpose priority queue.
+This priority queue module is implemented as a complete binary heap that manages elements with associated priorities.
+Each element in the priority queue has a priority, and the queue ensures that the element with the highest priority
+can be accessed in constant time.
+
+For example, to create a priority queue and manipulate its elements:
+<pre>
+struct priority_queue *pq;
+pq = priority_queue_create(10);
+
+int priority = 5;
+void *data = someDataPointer;
+
+priority_queue_push(pq, data, priority);
+data = priority_queue_pop(pq);
+void *headData = priority_queue_get_head(pq);
+</pre>
+
+To list all of the items in a priority queue, use a simple loop:
+<pre>
+for (int i = 1; i <= priority_queue_size(pq); i++) {
+    void *data = priority_queue_get_element(pq, i);
+    printf("Priority queue contains: %p\n", data);
+}
+</pre>
+
+Or use the PRIORITY_QUEUE_ITERATE macro:
+
+<pre>
+void *data;
+PRIORITY_QUEUE_ITERATE (pq, data) {
+    printf("Priority queue contains: %p\n", data);
+}
+</pre>
+*/
+
+
+/** Create a new priority queue.
+The 0th element is used as a sentinel with the highest priority to simplify boundary checks in heap operations like swim and sink.
```

Comment:
Is that a detail visible to the user of the structure, or just an implementation detail?

If it is visible, then you should probably say something about it in pop, iterate, etc.

If not, then move the comment to a good place in the .c file.